lib/benchee/formatters/console.ex

```diff
@@ -145,11 +155,65 @@ defmodule Benchee.Formatters.Console do
     units = Conversion.units(sorted_scenarios, scaling_strategy)
     label_width = label_width(sorted_scenarios)
 
-    [column_descriptors(label_width) |
+    output = [column_descriptors(label_width) |
       scenario_reports(sorted_scenarios, units, label_width)
       ++ comparison_report(sorted_scenarios, units, label_width, config)]
+
+    if config.extended_options do
+      output ++ [descriptor("Extended options") |
+        [extended_column_descriptors(label_width)]
+          ++ extended_options_reports(sorted_scenarios, units, label_width)]
+    else
+      output
+    end
+  end
+
+  @spec extended_options_reports([Scenario.t], map, integer) :: [any, ...]
+  defp extended_options_reports(scenarios, units, label_width) do
+    Enum.map(scenarios, fn(scenario) ->
+              format_scenario_extended(scenario, units, label_width) end)
+  end
+
+  @spec format_scenario_extended(Scenario.t, map, integer) :: String.t
+  defp format_scenario_extended(%Scenario{
+                                  job_name: name,
+                                  run_time_statistics: %Statistics{
+                                    minimum:     minimum,
+                                    maximum:     maximum,
+                                    sample_size: sample_size,
+                                    mode:        mode
+                                  },
+                                },
+                                %{run_time: run_time_unit},
+                                label_width) do
+    "~*s~*ts~*ts~*ts~*ts\n"
+    |> :io_lib.format([
+      -label_width, name,
+      @minimum_width, run_time_out(minimum, run_time_unit),
+      @maximum_width, run_time_out(maximum, run_time_unit),
+      @sample_size_width, to_string(sample_size),
+      @mode_width, mode_out(mode, run_time_unit)])
+    |> to_string
+  end
+
+  @spec mode_out([number], Benchee.Conversion.Unit.t) :: String.t
+  defp mode_out(modes, run_time_unit) when is_list(modes) do
+    Enum.map_join(modes, ", ", fn(mode) -> run_time_out(mode, run_time_unit) end)
+  end
+  defp mode_out(mode, run_time_unit) when is_number(mode) do
+    run_time_out(mode, run_time_unit)
+  end
```

Comment:
iirc this misses a case, `mode` can be nil if everything appears exactly the same amount of times (aka modeless). Also needs a test case :)

Reply:
Good catch! I've added a case where it just outputs the string "None" to be displayed. Is this sufficient or should we remove the mode column altogether maybe?

Reply:
Nope `None` is perfect :+1: